### ISSUE TITLE
log folder shall contain all files

### ISSUE THREAD

**Issue by test-fullautomation** [test-fullautomation/RobotFramework_AIO#135]
log folders for OSS and BIOS build shall contain all files generated by test during test-exeuction.
It shall also contain all console output.

**Comment by test-fullautomation** [comment]
Hi Thong,
please look into this.
Thank you,
 Thomas

**Comment by huavanthong** [comment]
Hi Thomas,  

Based on your comments, I reviewed all files generated by test during test execution. There are a lot of type of test will generated by each repository following table below:
| Repos \ Type of test | Atest  | Pytest | Pytest |
| --- | :---: |  :---: | :---: |
|  | aiotestlogfile_report.html  | aiotestlogfile.xml | log file for other test cases|
| python-extensions-collection |   | X |   |
| python-genpackagedoc |   | X |  |
| python-jsonpreprocessor | X | X |   |
| python-microservice-base |   | X |   |
| python-microservice-cleware-switch |   | X |  |
| python-pytestlog2db |   | X |   |
| tutorial-test |   | X |  |
| robotframework-extensions-collection |   | X |   |
| robotframework-robotlog2db |   | X |   |
| robotframework-robotlog2rqm |   | X |   |
| robotframework-testsuitesmanagement | X | X | X|

I noticed that only log files of test case in Pytest in **robotframework-testsuitesmanagement** has not yet been included in the log folder archive. And it will generated at directory **robotframework-testsuitesmanagement\test\testlogfiles**. 

Q1: I would like to confirm if you want to include that log file in the archive?

Q2:  Regarding all console output, I believe it's the console log displayed on the runner. Is my understanding correct?

**Comment by test-fullautomation** [comment]
Hi Thong,
very good!

Q1: yes, we should have all log output available. The idea is whatever it is it can help us later to analyze failures.
Q2: yes, correct. Perfect would be the whole console output which was created from triggering the first test, till the last test ending.

Thank you,
 Thomas

**Comment by huavanthong** [comment]
Hi Thomas, 
Cc Holger,

Thank you for your confirmation. I would like to raise my proposals after my investigation.
1. **About Q1:** 
* Because each repository will generate log file of test case in the different location, so I think it need to handle to move log fife as same as the process of Atest, Pytest ( aiotestlogfile.xml ).
* When I see [CConfig.py](https://github.com/test-fullautomation/robotframework-testsuitesmanagement/blob/a4330f6edf5c2cc6ca5e9b15093cc604dffb9797/test/libs/CConfig.py#L90) in robotframework-testsuitesmanagement, It recommend to use setting location of log file from external JSON config file. 
===> Holger, Could you think that it should handle on your side, then I will implement solution to get all log files to upload on artifacts as same as other log files? 

2. **About Q2:** 
* As my investigation, we can get all consoles output from Runner on Github Action, [refer](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/using-workflow-run-logs#downloading-logs).   
* And it's really details, It include log file for the entire runner, jobs inside runner, and steps inside job.
* Suppose if you want to see log file from triggering the first test, till the last test ending. You can download all log and go to "Test Windows package" -> "7_Run tests on Windows.txt" to see it.
===> I think it's enough log console for our information, if you want to upload all console output on our artifacts, I will implement it.

Let me know if you have any comments, 
Thank you,
Thong

**Comment by test-fullautomation** [comment]
Hi Thong,
Q1 must be discussed with holger.
Q2 the provided logfiles are very nice. Only we should not need to click, click, click till we download it. Instead it should be part of our logfile archive where we put all logs together. I'm sure there is an action which provides access to the log files.
The same we need for the gitlab build. I don't know the api there, it will be different to github?
Thank you,
 Thomas

**Comment by HolQue** [comment]
OOps, I just noticed that I have created a duplicate issue:

https://github.com/test-fullautomation/RobotFramework_AIO/issues/141

But you can refer to this issue, because I listed there the log files that are currently missing.

**Comment by HolQue** [comment]
Hello all,

how about this:

The test trigger configuration

`RobotFramework_AIO\test\aio-test-trigger\config\testtrigger_config.json`

already contains path and name of the log file required for the database tools. Example:

`"LOGFILE" : "../../../../python-jsonpreprocessor/atest/aiotestlogfiles/aiotestlogfile.xml"`

This file is already added to the downloadable log files archive.

I can add another (optional) key like:

`"FILES_SAVE" : "../../../../python-jsonpreprocessor/test/testlogfiles"`

This key is not for the test trigger (because the test trigger does not need this), but it's an information for the build process.

Maybe this key also need to be a semicolon separated list of paths.

Additionally to the already existing `"LOGFILE"`, now also the entire `"FILES_SAVE"` folder has to be added to the downloadable log files archive by the build process.

With this, we would have an entry in the test trigger configuration file, that is not used by the test trigger, but by a complete different process. But it belongs to the same topic (log files positions), and therefore I would say, it's OK.

**Comment by test-fullautomation** [comment]
@HolQue : what is the status of this task?

**Comment by HolQue** [comment]
Hi Thomas,

all Test Trigger configuration files have been extended by a key "FILES_SAVE" (where required) pointing to additional folders containing log files that also shall be collected. Now it's on the build process to use this information to collect these log files.

**Comment by test-fullautomation** [comment]
@huavanthong : What is the status for the build process?

**Comment by huavanthong** [comment]
Hi Thomas, 


**1.  About log files in AIO**

- I have updated the code to get all log files into log folder. (DONE)

**2.  About console logs on Runner**
-  I try to create a job to get console log in workflow by command :
```
gh run view <Running ID> --job <Specific job> --log
```

- However, there are an error with message below when I run command while pipeline is running, for a example:

```
$ gh run view 6193464302 --job 16815015206 --log
run 6193464302 is still in progress; logs will be available when it is complete
```
The error happened because the runner is still running - and not completed, so there are no log console for us.  And I found that this feature still not fixed in community, please refer: https://github.com/orgs/community/discussions/44250  

**Next steps** 
To work around this issues, I think I will focus on only **Step: Run test**. I will capture log when I trigger **aio-test-trigger.py** by redirect to a file. And upload it to artifacts

Please share your thoughts on this issues.

Thank you and best regards,  
Thong Hua

**Comment by test-fullautomation** [comment]
Hi Thong,
as discussed in our meeting: Please implement first the workaround (focus on Step: Run test).
Later we can see if we find a solution to capture and bring the whole console log to our log archive.
Thank you,
 Thomas

**Comment by huavanthong** [comment]
Hi Thomas,

I updated the implementation following our discussion. Currently, we can see the console log on artifacts at our zip file in directory **Robotframework_AIO/console_log.txt**.   
If there has any issues, please let me know.

Thank you and best regards,
Thong Hua

**Comment by test-fullautomation** [comment]
solved and merged with #147
